Shopify/cli
# Update the `app deploy` command for deployments release

### WHY are these changes introduced?

Fixes https://github.com/Shopify/app-deploys/issues/429



### WHAT is this pull request doing?

It does a few things to get the `app deploy` command ready for deployments release based on the mocks in the issue description (sorry for the scope creep!):
- Replaces the `createDeployment` mutation with the `appDeploy` mutation
- Adds support for a `--no-release` flag when running `app deploy` which, when used, skips releasing the created app version
- Adds a warning or info banner (depending on the unified deployment beta flag) that describes the changes being made to the `app deploy` command when it is run
- Updates the copy in the `app deploy` prompts and output info depending on the presence of the `--no-release` flag
- Updates the copy in the success banners (which also depend on the `--no-release` flag)
- Adds support for a general errors section (mainly for the case when an error is returned for a version that can't be released because it requires review)

<details><summary>Running `app deploy` when the deployments beta is disabled</summary>

![19-34-tmitz-avl0q](https://github.com/Shopify/cli/assets/65420305/3e819604-e366-44ab-9121-9b458de64fc6)

</details> 

<details><summary>Running `app deploy` when the deployments beta is enabled</summary>

##### `app deploy --no-release`
![create-only](https://github.com/Shopify/cli/assets/65420305/fb82df64-2cff-4f96-b90a-89e0e77f5173)

##### `app deploy`
![create-and-release](https://github.com/Shopify/cli/assets/65420305/696af4ea-44d0-479f-b639-286100c7d3d4)

##### `app deploy` when the deployment requires review
![release-error](https://github.com/Shopify/cli/assets/65420305/b7706712-2d0f-45a2-921c-0e6ba1f8ef33)

</details> 

### How to test your changes?

1. Create an app and generate an extension.
2. Run `app deploy` with the deployments beta disabled and verify that the extension is pushed to Shopify.
3. Enable the `app_unified_deployment` app beta flag and run `app deploy --no-release`. Verify that a deployment is created but **not** released.
4. Run `app deploy` and verify that a deployment can be both created **and** released.
5. Create an offsite payment provider extension (which requires approval) in the Partner dashboard; then, run `app deploy` and verify that a deployment is created but **not** released because of an error.

Note: Verify that the prompts, banners, and output info match the mocks while running through these flows.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
